tools/dev-doctor/binarycheck.go

```diff
@@ -79,15 +75,9 @@ func (c *binaryCheck) Run() (checkResult, string) {
 
 	if c.minVersion != nil {
 		// mangle the version we extracted into something that
-		// github.com/blang/semver will understand
+		// github.com/blang/semver/v4 will understand
 		versionBytes := origVersion
 
-		// sadly github.com/blang/semver.ParseTolerant is not tolerant of
-		// leading zeros in version numbers, as used by Docker, e.g. "19.03.13".
-		// so, remove leading zeros with a regexp. sorry, this is horrible but
-		// the alternative was bringing in a new version number handling module
-		versionBytes = leadingZerosRegexp.ReplaceAll(versionBytes, []byte("$1"))
-
 		version, err := semver.ParseTolerant(string(versionBytes))
```

Comment:
`versionBytes` can be dropped completely here - it's the same value as `origVersion` at this point. `origVersion` can then be renamed to just `version`.